nntrainer/layers/gru.cpp

```diff
@@ -159,26 +161,219 @@ void GRULayer::setRecurrentActivation(ActivationType activation) {
 }
 
 void GRULayer::forwarding(bool training) {
-  // NYI
+  Tensor &weight_xh =
+    weightAt(static_cast<int>(GRUParams::weight_xh)).getVariableRef();
+  Tensor &weight_hh =
+    weightAt(static_cast<int>(GRUParams::weight_hh)).getVariableRef();
+  Tensor &bias_h =
+    weightAt(static_cast<int>(GRUParams::bias_h)).getVariableRef();
+
+  hidden->getVariableRef().setZero();
+  zrg->getVariableRef().setZero();
+
+  h_prev.setZero();
+
+  Tensor &hidden_ = hidden->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  Tensor hs_prev;
+  Tensor hs;
+
+  for (unsigned int b = 0; b < input_dim[0].batch(); ++b) {
```

Comment:
it might be good to write the formula in comments here
that would also clarify zrg variables.

Reply:
done.